### PR TITLE
feat: Log a small percentage of non matching queries to Sentry

### PR DESCRIPTION
**Pull request by lynnagara**
Log 1 in 100 subscription queries that do not match with consistent
vs non-consistent to Sentry

### CODE REVIEW

File: snuba/subscriptions/worker.py [hunk 1]
```diff
@@ -180,6 +180,18 @@ def callback_func(
                         },
                     )
 
+                    # Only log 1 in 1000 mistmatches to sentry
+                    if random.random() < 0.001:
```
Comment (fpacifici) [inline snuba/subscriptions/worker.py:184]: Maybe log more aggressively, right now we are snuplicating only 5% of the queries. 50% of the transactions do not match so you would log 1/1000 of 50% of 5%. We can afford a higher sampling rate.

Comment (fpacifici) [reply snuba/subscriptions/worker.py:184]: Also this logs both matches and non matches.

File: snuba/subscriptions/worker.py [hunk 2]
```diff
@@ -180,6 +180,18 @@ def callback_func(
...
+                        logger.warning(
```
Comment (evanh) [inline snuba/subscriptions/worker.py:185]: I would also try to log the query itself in some form.

File: snuba/subscriptions/worker.py (hunk 1 above)
Comment (lynnagara) [reply snuba/subscriptions/worker.py]: Fixed

File: snuba/subscriptions/worker.py [hunk 3]
```diff
@@ -180,6 +180,19 @@ def callback_func(
...
+                    # Only log 1 in 100 mistmatches to sentry
+                    if match is False and random.random() < 0.01:
+                        logger.warning(
+                            "Non matching result with consistent and non-consistent subscription query",
+                            extra={
+                                "dataset": self.__dataset_name,
+                                "consistent": str(is_consistent_query),
+                                "primary": primary_result,
+                                "other": result.result,
+                                "query": request.query,
```
Comment (fpacifici) [inline snuba/subscriptions/worker.py:192]: This is the query object that does not have a pretty representation. I think you will just see the object id. 
If you want a formatted query you would need to pass this through a formatter as we were doing for the errors migration.

**Review by fpacifici — APPROVED** [review]
You may want to revisit what you are logging as the `query` object. Though not blocking for this as it is a minor issue we can fix later in the meantime we can start collecting data.

**Pull request merged**